### ISSUE TITLE
note: workspace flicker

### ISSUE THREAD

**Issue by salkin-mada** [johnae/persway#46]
@johnae regarding #42 #45 tested! It works. Thanks for the prompt fix.

An observation: the workspace seems to be _destroyed_ and a new one with the same name is _created_ just after. This results in a "flicker" of the workspace bar. Maybe its a question of refactoring? –  but would it not be ideal to have a behavior similar to `sway`'s own layout change, e.g. tabbed, stacked, split which does no such thing as a "pop/push" to the workspaces.

**Comment by johnae** [comment]
@salkin-mada Absolutely, that was what I initially looked into. It's non-trivial however as sway doesn't have a way to control layout other than moving windows up, down, left, right or to a mark. You have very little control unfortunately for something like master stack or layouts in general. Persway is, in some ways, essentially a hack.

As this issue (flickering etc) affects relayout only and I personally don't change layouts on the fly that often, it seemed like a reasonable tradeoff as we can reuse the way layouts otherwise are implemented.

I'm ofc not opposed to something better and if I find the time I might give this another try (I've tried several times already though and always ended up with worse tradeoffs unfortunately).
